Source/Functions.m

```diff
@@ -85,6 +94,29 @@
 	  NSLog (_(@"Cannot load the main model file '%@'"), mainModelFile);
 	}
     }
+  else
+    {
+      mainModelFile = [infoDict objectForKey: @"NSMainStoryboardFile"];
+      if (mainModelFile != nil && [mainModelFile isEqual: @""] == NO)
+        {
+          NSStoryboard *storyboard = [NSStoryboard storyboardWithName: mainModelFile
+                                                               bundle: [NSBundle mainBundle]];
+          if (storyboard == nil)
+            {
+              NSLog (_(@"Cannot load the main storyboard file '%@'"), mainModelFile);
+            }
+          else
+            {
+              [NSStoryboard _setMainStoryboard: storyboard];
```

Comment:
These three methods always get called together. Wouldn't it make more sense to have just one method here and have the rest over at NSStoryBoard?

Reply:
There is no documented method which does all three, so it would need to be private.  That's the only reason I didn't do it.

Reply:
You already use two undocumented methods here. This does not sound like a real argument.

Reply:
This code occurs once, so I don't think it's really necessary to do this now.

Reply:
The idea here was to have just the method _instantiateApplicationScene and in the implementation of that method call _setMainStoryboard:  and instantiateInitialController. Or if you would prefer to stick with the one public method have instantiateInitialController call the other two methods. The net result is always the sam, but we have less inter dependency and that by itself is a very good thing.

Reply:
OK, I just checked. The last option won't work because the method instantiateInitialController is also used for placeholders. So the best way is to hide everything behind _instantiateApplicationScene.